lib/puppet/provider/mysql.rb

```diff
@@ -24,7 +24,7 @@ def self.mysqld_type
     mysqld_version_string.scan(/mariadb/i) { return "mariadb" }
     mysqld_version_string.scan(/\s\(mysql/i) { return "mysql" }
     mysqld_version_string.scan(/\s\(percona/i) { return "percona" }
-    nil
+    return "mysql"
```

Comment:
It is better to match on what we know instead of assuming everything is "mysql"

add this above the `nil` instead

```
mysqld_version_string.scan(/\s\(\(ubuntu/i) { return "mysql" }
```